### PR TITLE
[RFR] Replace Upload binary test 

### PR DESCRIPTION
**Pull request by nachandr**
We don't always get consistent results with the current application being used for analysis. This has been reported in 
[Bug MTA-2606 ](https://issues.redhat.com/browse/MTA-2606).
So, I'm replacing the existing upload binary test with this PR. The new test uses the acmeair-webapp binary for analysis . 
The same app is  used in the UI tests as well.

### CODE REVIEW

File: analysis/tc_upload_binary.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+				RuleSet:     "cloud-readiness",
+				Rule:        "javax-to-jakarta-servlet-00100",
+			},
+		},
+	},
+	AnalysisTags: []api.Tag{
+		{Name: "Servlet", Category: api.Ref{Name: "HTTP"}},
+		{Name: "Servlet", Category: api.Ref{Name: "Java EE"}},
+		{Name: "Servlet", Category: api.Ref{Name: "Connect"}},
+	},
```
Comment (mguetta1) [inline analysis/tc_upload_binary.go:50]: Please check also analysis dependencies.
See example: https://github.com/konveyor/go-konveyor-tests/blob/main/analysis/tc_tackle_testapp_public_deps.go#L59

**Review by mguetta1 — CHANGES_REQUESTED** [review]
Thanks @nachandr for the PR!
Please see comments

**Review by aufi — CHANGES_REQUESTED** [review]
Thanks for PR, requesting changes mentioned by @mguetta1

File: analysis/tc_upload_binary.go (hunk 1 above)
Comment (nachandr) [reply analysis/tc_upload_binary.go:50]: Sure, the existing test didn't check analysis dependencies. But, I will add a check for that as well .

File: analysis/tc_upload_binary.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			{
+				Category:    "mandatory",
+				Description: "Method encodetUrl in javax.servlet.http.HttpServletResponse has been removed",
+				Effort:      1,
+				RuleSet:     "cloud-readiness",
+				Rule:        "javax-to-jakarta-servlet-00100",
+			},
+		},
+	},
+	AnalysisTags: []api.Tag{
```
[the remark this answers is not in the document]
Comment (nachandr) [reply analysis/tc_upload_binary.go:46]: Yes, I noticed that on Friday.  It's strange that the UI shows 3 tags. I have to investigate why there is a mismatch.

**Comment by nachandr** [comment]
On MTA 7.0.3-25, I noticed that 'Effort' and 'Tags' show zero after successful analysis of binary applications .
I reported [bug MTA-2821](https://issues.redhat.com/browse/MTA-2821) yesterday so that issue could be addressed.

**Comment by nachandr** [comment]
I continue to see inconsistent values for Effort, Tag count, Issue count, Dependency count between upstream and downstream builds.

I did the development and testing on d/s. 

Additionally, I see inconsistent values when app analysis is done on UI and through API on d/s build.

**Comment by mguetta1** [comment]
Thanks @nachandr 
Please move it to TIER2 for now

**Comment by nachandr** [comment]
Results from PR tester run
1) I added a new TIER0 test and all the tier0 tests are passing
2) One of the TIER2 tests is failing because of [bug MTA-2606](https://issues.redhat.com/browse/MTA-2606) ,

**Review by aufi — APPROVED** [review]
LGTM

**Review by mguetta1 — APPROVED** [review]
LGTM

**Pull request merged**